### PR TITLE
HW2 submitted by Sean20405 (Attempt 1)

### PR DESCRIPTION
**Pull request by Sean20405**
In this PR, I've finished HW2, including reimplement class `Line` (Q1) and calculate angle as well as using `pybind11` and `pytest` (Q2).

---

A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: hw2/Sean20405/validate.sh [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  and rebuild.
+* ``make test`` runs the Python-based tests.
+* ``make clean`` removes all the built and intermediate files.
+* To get full point one has to do all 3.
+EOF
+echo "GET POINT 1"
+
+rm -rf $tmp_dir
+
+# vim: set fenc=utf8 ff=unix et sw=2 ts=2 sts=2:
```
Comment (ThreeMonth03) [inline hw2/Sean20405/validate.sh]: Don't submit the validate script.

File: hw2/Sean20405/golden.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+line: number of points = 3
+point 0: x = 0 y = 1
+point 1: x = 1 y = 3
+point 2: x = 2 y = 5
+line2: number of points = 3
+point 0: x = 9 y = 1
+point 1: x = 1 y = 3
+point 2: x = 2 y = 5
```
Comment (ThreeMonth03) [inline hw2/Sean20405/golden.txt]: Don't submit golden text.

**Review by ThreeMonth03 — CHANGES_REQUESTED** [review]
- [ ] Please remove the unnecessary files.

**Comment by Sean20405** [comment]
OK. I've removed them!

**Review by ThreeMonth03 — APPROVED** [review]
Thanks, the pull request looks better now.

**Pull request merged**